AtmosphereAutopilot/Modules/ProgradeThrustController.cs

```diff
@@ -40,6 +40,96 @@ public struct SpeedSetpoint
         public const float kts2mps = 0.514444f;
         public const float mps2kts = 1.0f / kts2mps;
 
+        public const float machEpsilon = 0.005f;
+
+        public static float rayleighPitotTubeStagPressureSubSonic(float M, float gamma)
+        {
+            float pressure;
+
+            pressure = (M * M) * (gamma - 1) / 2 + 1;
+            pressure = Mathf.Pow(pressure, gamma / (gamma - 1));
+
+            return pressure;
+        }
+
+        public static float rayleighPitotTubeStagPressureSuperSonic(float M, float gamma)
+        {
+            float pressure;
+
+            pressure = (gamma + 1) * M;
+            pressure *= pressure;
+            pressure /= (4 * gamma * M * M - 2 * (gamma - 1));
+            pressure = Mathf.Pow(pressure, gamma / (gamma - 1));
+            pressure *= (1 - gamma + 2 * gamma * M * M) / (gamma + 1);
+
+            return pressure;
+        }
+
+        public static float mps2Ias(float mps, Vessel v)
+        {
+            float M = (float)(mps / v.speedOfSound);
```

Comment:
For every division, sqrt and other possible NaN-poisoning operations, please check that we don't fall flat.
Vacuum or some other conditions may break the new code - you should probably handle it just in case.